features/step_definitions/element_steps.rb

```diff
@@ -55,16 +55,23 @@
 
 Then('all mapped elements are present') do
   mapped_item_names =
-    @test_site.dynamic.class.mapped_items.map(&:values).flatten.sort
+    @test_site.dynamic.class.mapped_items.map(&:values).flatten
 
-  expect(@test_site.dynamic.elements_present.sort).to eq(mapped_item_names)
+  expect(@test_site.dynamic.elements_present).to match_array(mapped_item_names)
 end
 
 Then('not all mapped elements are present') do
   mapped_item_names =
     @test_site.dynamic.class.mapped_items.map(&:values).flatten
 
-  expect(@test_site.dynamic.elements_present).not_to eq(mapped_item_names)
+  expect(@test_site.dynamic.elements_present).not_to match_array(mapped_item_names)
+end
+
+Then('missing elements are returned') do
+  mapped_item_names =
+    @test_site.dynamic.class.mapped_items.map(&:values).flatten
+
+  expect(@test_site.dynamic.missing_elements).to match_array(mapped_item_names)
```

Comment:
this doesn't make sense to me. Because the dynamic page has a bunch of elements that are present. So surely the missing elements on dynamic page should be an empty array? Or am I missing something?

Reply:
I used the same approach as the negative test above (I'd not say it's good) [here](https://github.com/site-prism/site_prism/pull/52/files#diff-1f9996f18a011c66b2e4de840767086eR72) and [here](https://github.com/site-prism/site_prism/pull/52/files#diff-5e497d00aedf9e13bd7fefa6a06e8126R63). 
[Here](https://github.com/site-prism/site_prism/pull/52/files#diff-1f9996f18a011c66b2e4de840767086eR6) in the background we navigate to the home page but checking the dynamic page elements

Reply:
No I understand the naming, just the outcome seems odd. Dynamic page should have *no* missing elements. So the fact you're testing this page with missing elements feels odd to me.

Reply:
Rebase with master, and then hopefully I can re-review and we can look into merging this @ineverov

Reply:
@luke-hill rebased and fixed